src/Microsoft.DotNet.Tools.Publish/Program.cs

```diff
@@ -50,18 +54,28 @@ public static int Main(string[] args)
                     path = Directory.GetCurrentDirectory();
                 }
 
-                // Load project context and publish it
-                var fx = NuGetFramework.Parse(framework.Value());
-                var rids = new[] { runtime.Value() };
-                var context = ProjectContext.Create(path, fx, rids);
+                var projectContexts = ProjectContext.CreateContextForEachTarget(path);
+                projectContexts = GetMatchingProjectContexts(projectContexts, framework.Value(), runtime.Value());
```

Comment:
This looks inefficient if you aren't aware that we're planning the ability to publish multiple targets. Maybe add a comment to ensure nobody goes in an "helpfully" re-optimizes this? ;P